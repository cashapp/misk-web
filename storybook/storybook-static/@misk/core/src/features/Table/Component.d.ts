export interface ITableProps {
    data: any[];
    range?: number[];
}
export declare const Table: (props: ITableProps) => JSX.Element;
