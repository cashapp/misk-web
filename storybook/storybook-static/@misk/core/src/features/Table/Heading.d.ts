export declare const Heading: (props: {
    data: any;
}) => JSX.Element;
