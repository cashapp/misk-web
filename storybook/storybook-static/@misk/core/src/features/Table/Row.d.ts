export declare const Row: (props: {
    data: any;
    index?: number;
}) => JSX.Element;
