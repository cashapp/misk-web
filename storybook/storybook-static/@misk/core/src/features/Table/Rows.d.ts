export declare const Rows: (props: {
    data: any;
    range: number[];
}) => JSX.Element;
