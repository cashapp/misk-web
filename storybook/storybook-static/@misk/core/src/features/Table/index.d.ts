export * from "./Component";
export * from "./Heading";
export * from "./Row";
export * from "./Rows";
