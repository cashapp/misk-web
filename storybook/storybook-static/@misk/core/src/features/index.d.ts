export * from "./Navbar";
export * from "./Table";
