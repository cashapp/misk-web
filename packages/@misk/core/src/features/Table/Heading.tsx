import * as React from "react"

export const Heading = (props: { data: any }) => {
  const { data } = props
  return (
    <thead>
      <tr>
        <th key={"index"}>{""}</th>
        {Object.entries(data).map(([k]) => (
          <th key={k}>{k}</th>
        ))}
      </tr>
    </thead>
  )
}
