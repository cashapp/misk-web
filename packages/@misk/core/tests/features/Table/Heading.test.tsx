import React from "react"
import { cleanup, render } from "@testing-library/react"
import { Heading } from "src/features/Table"
import { cars } from "./testFixtures"

describe("Table Heading", () => {
  afterEach(cleanup)
  it("Heading can render", () => {
    const { asFragment } = render(<Heading data={cars[0]} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
