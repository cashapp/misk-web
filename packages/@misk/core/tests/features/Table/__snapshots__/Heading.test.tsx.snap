// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Heading Heading can render 1`] = `
<DocumentFragment>
  <thead>
    <tr>
      <th />
      <th>
        id
      </th>
      <th>
        make
      </th>
      <th>
        model
      </th>
      <th>
        model_year
      </th>
      <th>
        quantity
      </th>
      <th>
        price
      </th>
      <th>
        slogan
      </th>
      <th>
        latitude
      </th>
      <th>
        longitude
      </th>
      <th>
        manufactured_date
      </th>
      <th>
        created_at
      </th>
      <th>
        updated_at
      </th>
    </tr>
  </thead>
</DocumentFragment>
`;
